maze_transformer/generation/lattice_maze.py

```diff
@@ -137,14 +144,22 @@ def get_coord_neighbors(self, c: Coord) -> CoordArray:
             )
         ]
 
-        return np.array(neighbors)
+        output: CoordArray = np.array(neighbors)
+        if len(neighbors) > 0:
+            assert output.shape == (
```

Comment:
What are we guarding against here?

Reply:
neighbors is initialized as a list, since we don't know the length of it to start with. many of these functions should be vectorized and converted to pure numpy, but thats out of scope for this PR. However, the output of `get_coord_neighbors` is expected to be a numpy array